### PR TITLE
button link housekeeping

### PR DESCRIPTION
**Pull request by garrettbear**
## Explanation

* Fixes #16635 

### Before
Screenshot 2022-12-16 at 10 17 32 AM

### After
Screenshot 2022-12-16 at 10 18 19 AM

## Manual Testing Steps
`yarn test:unit:jest .ui/components/component-library/button-link/button-link.test.js`

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [x] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [x] PR is linked to the appropriate GitHub issue
- [] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### Housekeeping
Ensure that `ButtonLink` adheres to all of the following conventions and standards

- [x] Has a `className` prop and the PropType descriptions are all the same
- [x] Prop table in MDX docs have the "Accepts all Box component props" description and link
- [ ] We are consistent when using the same prop names like `size` and are suggesting the use of the generalized `design-system.js` constants e.g. `SIZES` as the primary option but noting the component consts in the documentation and using them for propType validation and storybook controls only
- [x] Standardize all similar prop names for images `imgSrc`, `imgAlt`(html element + attribute) (needs audit)
- [x] We have a story for each component prop and we use the prop name verbatim e.g. `size` prop would be `export const Size = (args) => (`
- [x] We have the accompanying documentation for each component prop and we use the prop name verbatim e.g. `size` prop would be `### Size`
- [x] Are multiple props stories allowed? e.g. `Color, Background Color And Border Color` story in `base-avatar` - [ ] yes when it makes sense to
- [ ] All Base components follow the suffix convention e.g. `ButtonBase`
- [ ] All Base component MDX documentation have the base component notification at the top
- [x] Add `mm-` prefix to all classNames
- [x] className is kebab case version of the component name
- [x] Spread base components props and reduce duplication of props when props aren't being changed and remain the same for both variant and base components
- [x] Add component to root `index.js` file in component-library
- [ ] Add locals for any default text I18nContext as default context 
- [ ] Add any "to dos" with a `// TODO:` comment so we can search for them at a later date e.g. blocking components etc
- [x] Add snapshot testing
- [x] Add pixel values to propType descriptions if we use abstracted prop types that relate to pixel values e.g. `SIZE.MD (32px)`
- [x] Each prop section in the MDX docs should have: a heading, a description, a story and an example code snipped

### CODE REVIEW

File: ui/components/component-library/button-base/button-base.js [hunk 1]
```diff
@@ -36,8 +36,8 @@ export const ButtonBase = ({
     <Box
       as={Tag}
       href={href}
-      paddingLeft={size === BUTTON_BASE_SIZES.AUTO ? 0 : 4}
-      paddingRight={size === BUTTON_BASE_SIZES.AUTO ? 0 : 4}
+      paddingLeft={size === BUTTON_BASE_SIZES.INHERIT ? 0 : 4}
+      paddingRight={size === BUTTON_BASE_SIZES.INHERIT ? 0 : 4}
```
Comment (georgewrmarshall) [inline ui/components/component-library/button-base/button-base.js:40]: Should SIZES.INHERIT only concern `ButtonLink`?

**Review by georgewrmarshall — COMMENTED** [review]
Looking good. Couldn't find any instances of `SIZES.AUTO` in the code base ✔️. Should the inherit size option only concern `ButtonLink`?

One small update to icon docs. Remove "`Auto also exists and`" in two places. 

Screenshot 2022-12-19 at 1 45 22 PM

**Review by georgewrmarshall — COMMENTED** [review]
I know we've been going over this quite a lot and I've made a lot of suggestions already. So my sincere apologies for dragging this out but I tried an alternative exploration in this PR https://github.com/MetaMask/metamask-extension/pull/17034. I've updated all the documentation and stories so "if" we decide to move forward with it, it shouldn't be any extra work after merge.

File: ui/components/ui/box/box.stories.js [hunk 2]
```diff
@@ -299,7 +299,7 @@ export const Padding = (args) => {
       </Box>
       <Box
         {...args}
-        padding={[4, 8, 12]}
+        padding={[4, 0, 8, 12]}
```
Comment (georgewrmarshall) [inline ui/components/ui/box/box.stories.js:302]: Not sure if this is supposed to be here but no big deal

**Review by georgewrmarshall — DISMISSED** [review]
Love it 🔥

File: ui/components/component-library/button-link/README.mdx [hunk 3]
```diff
@@ -16,31 +17,57 @@ The `ButtonLink` accepts all props below as well as all [Box](/docs/components-u
 
 <ArgsTable of={ButtonLink} />
 
+The `ButtonLink` accepts all [ButtonBase](/docs/ui-components-component-library-button-base-button-base-stories-js--default-story#props) component props
```
Comment (NidhiKJha) [inline ui/components/component-library/button-link/README.mdx:20]:
```suggestion
The `ButtonLink` accepts all [ButtonBase](/docs/components-componentlibrary-buttonbase--default-story#props) component props
```

**Review by NidhiKJha — CHANGES_REQUESTED** [review]
Can we update the ButtonBase link with the new url

**Review by georgewrmarshall — COMMENTED** [review]
Changes looking good just need to resolve conflicts

**Review by NidhiKJha — DISMISSED** [review]

**Review by georgewrmarshall — APPROVED** [review]

**Review by NidhiKJha — APPROVED** [review]

**Pull request merged**